erasure-coding/src/lib.rs

```diff
@@ -124,8 +125,13 @@ fn code_params(n_validators: usize) -> Result<CodeParams, Error> {
 /// Obtain erasure-coded chunks, one for each validator.
 ///
 /// Works only up to 65536 validators, and `n_validators` must be non-zero.
-pub fn obtain_chunks(n_validators: usize, block_data: &BlockData, outgoing: &OutgoingMessages)
-	-> Result<Vec<Vec<u8>>, Error>
+pub fn obtain_chunks(
+	n_validators: usize,
+	relay_parent: H256,
+	candidate_hash: H256,
```

Comment:
not a huge fan of leaking the `relay_parent` and `candidate_hash` into this function, which was relatively self-contained before. I feel that this logic and the `branches` call were meant to be called by some higher level function that combines them.

There's also a bit of weird circular logic in the interface here:
  1. the `Candidate` struct should have a field for the availability-trie root hash so nodes have something to verify against. That should be added as part of this PR.
  2. so creating the trie and the chunks has to be done before creating the `Candidate`
  3. this interface for the function to create the trie and the chunks requires the `Candidate` hash, which can only be known after creating the candidate